### DISCUSSION TITLE
Executing synchronous (non-async) scenarios concurrently

### DISCUSSION THREAD

**Discussion by kevgo** [cucumber-rs/cucumber#279] [Q&A] [answered]
I prefer writing my step implementations as synchronous code (instead of async). Example:

```rs
use std::time::Duration;

#[given("a satiated cat")]
fn satiated_cat(world: &mut AnimalWorld) {
    std::thread::sleep(Duration::from_millis(1000));
    world.cat.hungry = false;
}
```

However, doing so executes all scenarios sequentially. I assume that's because Cucumber-Rust works asynchronously internally and my sync step implementations block the event loop. Is it possible to configure Cucumber-Rust to execute my Scenarios with sync step implementations concurrently?

**Comment by ilslv** [comment, marked as answer]
@kevgo you can make your sync functions async by using [`tokio::spawn_blocking`](https://docs.rs/tokio/latest/tokio/task/fn.spawn_blocking.html)/[`tokio::block_in_place`](https://docs.rs/tokio/latest/tokio/task/fn.block_in_place.html). You may also want to check something like [`async_scoped`](https://docs.rs/async-scoped/) and [`moro`](https://docs.rs/moro/) for [`thread::scope`](https://doc.rust-lang.org/stable/std/thread/fn.scope.html)-like API. We would like to do it ourselves, but unfortunately currently there is no runtime-agnostic way to do it.
